Modules/Server/Sources/PocketCastsServer/Public/Helpers/URLRequest+RefreshIfNeeded.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+public extension URLRequest {
+
+    mutating func setEtagAndLastModifiedHeaders(cachedResponse: CachedURLResponse) {
+        if let etag = cachedResponse.response.etag {
```

Comment:
While testing different podcasts I found an issue that is not directly related to this PR but it will affect this:

http://www.stephen-gurnett.org/unlikely-objects/2017/2/1/fylkgu09vo4g3zhofzyv9gucz64alt

Basically, when testing some podcasts (like "If Books Could Kill") cache never worked, even though there's a `etag` there. The issue is that it's written `Etag` not `ETag` as we expect.

I believe we should implement this fix so we don't run in cases like this.

Not in this PR though.

Reply:
So the plan is another PR is to try to fetch the elements by using multiple possible variations of the tag? I wonder if we could copy the dictionary and lowercase all the keys?